lib/DxilValidation/DxilContainerValidation.cpp

```diff
@@ -73,6 +78,523 @@ struct DiagRestore {
+    DMPSV.UsesViewID = DM.m_ShaderFlags.GetViewID() ? true : false;
+    DMPSV.SigInputVectors = DM.GetInputSignature().NumVectorsUsed(0);
+    for (unsigned streamIndex = 0; streamIndex < 4; streamIndex++)
+      DMPSV.SigOutputVectors[streamIndex] =
+          DM.GetOutputSignature().NumVectorsUsed(streamIndex);
+    if (SM->IsHS() || SM->IsDS() || SM->IsMS())
+      DMPSV.SigPatchConstOrPrimVectors =
+          DM.GetPatchConstOrPrimSignature().NumVectorsUsed(0);
+  }
+  bool Mismatched = false;
+  if (PSV2)
+    Mismatched = memcmp(PSV2, &DMPSV, sizeof(PSVRuntimeInfo2)) != 0;
+  else if (PSV1)
+    Mismatched = memcmp(PSV1, &DMPSV, sizeof(PSVRuntimeInfo1)) != 0;
+  else
+    Mismatched = memcmp(PSV0, &DMPSV, sizeof(PSVRuntimeInfo0)) != 0;
+
+  if (Mismatched) {
+    std::string Str;
+    raw_string_ostream OS(Str);
+    hlsl::PrintPSVRuntimeInfo(OS, &DMPSV, &DMPSV, &DMPSV, &DMPSV,
+                              static_cast<uint8_t>(SM->GetKind()),
+                              DM.GetEntryFunctionName().c_str(), "");
+    OS.flush();
+    std::string Str1;
+    raw_string_ostream OS1(Str1);
+    PSV.PrintPSVRuntimeInfo(OS1, static_cast<uint8_t>(PSVShaderKind::Library),
+                            "");
+    OS1.flush();
+    EmitMismatchError("PSVRuntimeInfo", Str, Str1);
+  }
+}
+
+void PSVContentVerifier::Verify() {
+  unsigned ValMajor, ValMinor;
+  DM.GetValidatorVersion(ValMajor, ValMinor);
+  unsigned PSVVersion = hlsl::GetPSVVersion(ValMajor, ValMinor);
+
+  PSVInitInfo PSVInfo(PSVVersion);
+  if (PSV.GetRuntimeInfoSize() != PSVInfo.RuntimeInfoSize()) {
+    EmitMismatchError("PSVRuntimeInfoSize",
+                      std::to_string(PSV.GetRuntimeInfoSize()),
+                      std::to_string(PSVInfo.RuntimeInfoSize()));
+    return;
+  }
+
+  if (PSV.GetBindCount() > 0 &&
+      PSV.GetResourceBindInfoSize() != PSVInfo.ResourceBindInfoSize()) {
+    EmitMismatchError("ResourceBindInfoSize",
+                      std::to_string(PSV.GetResourceBindInfoSize()),
+                      std::to_string(PSVInfo.ResourceBindInfoSize()));
+    return;
+  }
+  VerifyResources(PSVVersion);
+
+  PSVRuntimeInfo0 *PSV0 = PSV.GetPSVRuntimeInfo0();
+  PSVRuntimeInfo1 *PSV1 = PSV.GetPSVRuntimeInfo1();
+  PSVRuntimeInfo2 *PSV2 = PSV.GetPSVRuntimeInfo2();
+
+  const ShaderModel *SM = DM.GetShaderModel();
+  VerifyEntryProperties(SM, PSV0, PSV1, PSV2);
+  if (PSVVersion > 0) {
+    if (((PSV.GetSigInputElements() + PSV.GetSigOutputElements() +
+          PSV.GetSigPatchConstOrPrimElements()) > 0) &&
+        PSV.GetSignatureElementSize() != PSVInfo.SignatureElementSize()) {
+      EmitMismatchError("SignatureElementSize",
+                        std::to_string(PSV.GetSignatureElementSize()),
+                        std::to_string(PSVInfo.SignatureElementSize()));
+      return;
+    }
+    uint8_t ShaderStage = static_cast<uint8_t>(SM->GetKind());
+    PSVRuntimeInfo1 *PSV1 = PSV.GetPSVRuntimeInfo1();
```

Comment:
We already have `PSV1` set above, (line 555) so don't need to shadow that variable here, or `VerifyEntryProperties` could just take `PSV` and get the versioned structs itself.

Reply:
Removed.